docs/source/developers/index.md

```diff
@@ -0,0 +1,19 @@
+# Developers
+
+The developer documentation is for authors who want to enhance
+the functionality of Jupyter AI.
+
+If you are interested in contributing to Jupyter AI,
+please see our {doc}`contributor's guide </contributors/index>`.
+
+## Pydantic compatibility
+
+Jupyter AI internally uses **Pydantic v1** and should work with either Pydantic
+version 1 or version 2. For compatibility, developers using Pydantic version 2
+should import classes using the `pydantic.v1` package. See the
+[LangChain Pydantic migration plan](https://python.langchain.com/docs/guides/pydantic_compatibility)
+for advice about how developers should use `v1` to avoid mixing v1 and v2
+classes in their code.
+
+If you build an application that depends on LangChain, you can import from
+`langchain.pydantic_v1`, which exposes Pydantic v1 classes from both Pydantic v1 and v2.
```

Comment:
I don't think this is worded clearly.

> Jupyter AI internally uses **Pydantic v1** and should work with either Pydantic
version 1 or version 2.

This is vague; what does "should work with" mean here? We should say something more along the lines of:

"Jupyter AI is fully compatible with Python environments using Pydantic v1 and those using Pydantic v2. Jupyter AI does so by not having a direct dependency on `pydantic`. Instead, Jupyter AI imports Pydantic via the `langchain.pydantic_v1` module. Developers should do the same when extending Jupyter AI classes."

> For compatibility, developers using Pydantic version 2
should import classes using the `pydantic.v1` package. See the
[LangChain Pydantic migration plan](https://python.langchain.com/docs/guides/pydantic_compatibility)
for advice about how developers should use `v1` to avoid mixing v1 and v2
classes in their code.

I think this section can be mostly excluded. We don't need to explain how you can access the v1 API in both Pydantic v2. All the developer needs to know is "use `langchain.pydantic_v1` instead of `pydantic` when developing with Jupyter AI".

At most, we can drop a quick reference, like "For more details, see the LangChain documentation on Pydantic compatibility."

> If you build an application that depends on LangChain, you can import from
`langchain.pydantic_v1`, which exposes Pydantic v1 classes from both Pydantic v1 and v2.

The "if" clause is redundant. If Jupyter AI is installed, then `langchain` will also be installed. So we can just recommend developers use `langchain.pydantic_v1` when extending our classes.